src/Image/Composite/HelioviewerCompositeImage.php

```diff
@@ -594,11 +597,22 @@ private function _addEventLayer($imagickImage) {
 
         require_once HV_ROOT_DIR.'/../src/Event/HEKAdapter.php';
         require_once HV_ROOT_DIR.'/../src/Database/FlarePredictionDatabase.php';
+        require_once HV_ROOT_DIR.'/../scripts/rot_hpc.php';
 
         // Collect events from all data sources.
         $hek = new Event_HEKAdapter();
         $event_categories = $hek->getNormalizedEvents($this->date, Array());
         $event_categories = array_merge($event_categories, Database_FlarePredictionDatabase::GetLatestNormalizedFlarePredictions($this->date));
+        $startDate = new DateTimeImmutable($this->date);
+        $endDate = $startDate->add(new DateInterval("P1D"));
+        $observationDate = $this->date;
+        $applyRotation = function ($hv_event) use ($observationDate) {
+            // Apply solar rotation from the event time to the current observation time
+            list($hv_event->hv_hpc_x, $hv_event->hv_hpc_y) = rot_hpc($hv_event->hpc_x, $hv_event->hpc_y, $hv_event->start, $observationDate);
+            return $hv_event;
+        };
+        $tmp = Events::GetAll($startDate, $endDate, $applyRotation);
```

Comment:
This is duplicated, should be moved into a helper so it can be done in SolarEvents.php and here.